pallets/msa/src/lib.rs

```diff
@@ -610,12 +610,18 @@ pub mod pallet {
 			Ok(())
 		}
 
-		/// Adds a given `new_key` to `msa_id` of the account signing `msa_owner_proof`, which must match the MSA in `add_key_payload`.
-		/// The `new_key` must sign the `add_key_payload` to authorize the addition.
+		/// Adds a new public key to an existing Message Source Account (MSA). This functionality enables the MSA owner to manage multiple keys
+		/// for their account or rotate keys for enhanced security.
 		///
-		/// # Remarks
-		/// * Origin can be same as msa owner.
-		/// * Signatures should be over the [`AddKeyData`] struct
+		/// The `origin` parameter represents the account from which the function is called and can be either the MSA owner's account or a delegated provider's account,
+		///  depending on the intended use.
+		///
+		/// The function requires two signatures: `msa_owner_proof` and `new_key_owner_proof`, which serve as proofs of ownership for the existing MSA
```

Comment:
only `msa_owner_proof` serves as a proof of ownership for the existing MSA.  `new_key_owner_proof` proves that the submitter of the extrinsic got the new key owner to sign the payload.  It's really that the extrinsic submitter can't add a new key without both the MSA owner, AND the new key owner's permission.

Reply:
The critical word is "respectively" which separates the two.  The overall intent of getting the permission of both the owner of the MSA and the owner of the new key will be added.